### DISCUSSION TITLE
ZOS 3.0 AppService httpRequest (bug?)

### DISCUSSION THREAD

**Discussion by AFumi39** [zepp-health/zeppos-forum#285] [Developer Feedback]
Hello to everyone, as stated in a previous discussion (https://github.com/orgs/zepp-health/discussions/270) I was trying to send fitness data, at timed intervals, to an external WebService.

I managed to develop an easy MiniApp that use ZML Library (https://github.com/zepp-health/zml) with a Page that have 2 buttons that works in this way:
- Button 1 send data when pressed
- Button 2 start/stop the AppService

The AppService have a timeSensor that invoke the same httpRequest every 5 minutes (using onPerMinute method with a `if(timeSensor.getMinutes() % 5 == 0)`.

There's a big problem: if I woke up the watch (turn the screen on) within the minute when the AppService have to send data, all works as expected. BUT if I don't turn the screen on, the httpRequest remains "stuck" and no other httpRequest can be made (by the AppService nor the Page). Even the Page does not works, but no error returned, simply do nothing.
To make it works again I have to stop the AppService, close the MiniApp, the open the MiniApp again, so the Button 1 resume it's expected behaviour.

Do you have an idea to make it works?

Thank you

**Comment by MyoungXUE** [comment]
Sorry, this restriction is not clearly stated in the documentation, I will make adjustments ASAP.

You need to use the AOD watchface to ensure that the screen is running so that network requests can be sent normally.

Watch settings -> Display and brightness -> Always-on display

**Reply by AFumi39** [reply]
Thank you, I'll try that 👍

**Reply by AFumi39** [reply]
I confirm that now (with AOD) works 🎉
But if for some reason it can't do the httpRequest (maybe network problem, maybe else), the "stuck" problem of ZML occurs again, and I have to kill the service and the app, and then restart again. 🤔

**Reply by htoooth** [reply]
@AFumi39 You should use the latest zml lib. I have fixed some bugs, such as memory leak.

**Reply by AFumi39** [reply]
@htoooth thanks!
